cylc/flow/task_pool.py

```diff
@@ -119,6 +121,51 @@
 Pool = Dict['PointBase', Dict[str, TaskProxy]]
 
 
+def _standardise_xtrigs(
+    prereqs: 'Iterable[str]'
+) -> 'Dict[str, bool]':
+    """Extract xtriggers from user input and standardise.
+
+    Weed out any task prerequisites.
+
+    Note command validation has already failed illegal state suffixes,
+    and added the default ":succeeded".
+
+    Standardisation handles ":succeed" -> ":succeeded".
+
+    Args:
+        prereqs: prerequisites and xtriggers in string form
+            xtriggers format "xtrigger/label:state"
+
+    Returns: {label: (succeeded, set-output)}
+        (set-output means set the xtrigger output for all dependent tasks)
+
+    Examples:
+        >>> _standardise_xtrigs(["1/foo:started"])
```

Comment:
MIght be nice to see some examples using other types of iterable, just to be on the safe side?

Reply:
OK I converted one to a set.